### PR TITLE
[Fix] Remove Trailing Whitespaces from Char Column in Postgresql/Redshift Connector

### PR DESCRIPTION
**Pull request by Jithendar12**
*Issue #, if available:*
[ [BUG] Redshift 'char' column constraint pushdown does not work #62 ](https://github.com/awslabs/aws-athena-query-federation/issues/62)

*Description of changes:*
PostgreSQL/Redshift database will send data along with trailing whitespaces in CHAR datatype columns. So, removed those spaces using the RTRIM() function while building the SQL query.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java [hunk 1]
```diff
@@ -445,4 +446,31 @@ protected PreparedStatement getMaterializedViewOrExternalTable(Connection connec
...
+    /**
+     * Retrieves the names of columns with the data type 'CHAR' for a specified table in a PostgreSQL/Redshift database.
+     *
+     * @param connection the JDBC connection to the database
+     * @param schema Postgresql/Redshift schema name
+     * @param table Postgresql/Redshift table name
+     * @return a list of column names that have the data type 'CHAR'
+     * @throws SQLException if a database access error occurs
+     */
+    public static List<String> getCharColumns(Connection connection, String schema, String table) throws SQLException
```
Comment (aimethed) [inline athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java:459]: Why is this method in the metadata handler and not where it is called in the query string builder?

File: athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java [hunk 2]
```diff
@@ -445,4 +446,31 @@ protected PreparedStatement getMaterializedViewOrExternalTable(Connection connec
...
+    {
+        List<String> charColumns = new ArrayList<>();
+        String query = "SELECT column_name " +
+                "FROM information_schema.columns " +
+                "WHERE table_schema = ? AND table_name = ? AND data_type = 'character'";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, schema);
+            statement.setString(2, table);
+            try (ResultSet resultSet = statement.executeQuery()) {
```
Comment (aimethed) [inline athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java:468]: Do we need to make an additional call to the postgresql database to determine this? Is the character datatype gotten from this query different from the datatype we can get from the tableSchema? An apache arrow type has the FieldType for each field in the arrow schema. Is there a case where the FieldType might not align with the type gotten from the postgresql database?

**Review by aimethed — CHANGES_REQUESTED** [review]

File: athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java (hunk 1 above)
Comment (Jithendar12) [reply athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java:459]: Hi @aimethed , as other metadata related methods are present in this class so added it here.

File: athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java (hunk 2 above)
Comment (Jithendar12) [reply athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java:468]: Hi @aimethed , yes we make an additional call to postgresql to retreive fixed-length 'char' columns to trim any extra trailing spaces in the data. Yes the datatype obtained from the tableSchema is different which is 'varchar'. Yes it seems apache arrow type does not have a specific data type for fixed-length 'char'.

**Review by aimethed — APPROVED** [review]

**Comment by AbdulR3hman** [comment]
This has been addressed in the latest release.

**Pull request merged**